src/network/networkd.c

```diff
@@ -62,7 +62,9 @@ static int run(int argc, char *argv[]) {
                                     (1ULL << CAP_NET_ADMIN) |
                                     (1ULL << CAP_NET_BIND_SERVICE) |
                                     (1ULL << CAP_NET_BROADCAST) |
-                                    (1ULL << CAP_NET_RAW));
+                                    (1ULL << CAP_NET_RAW) |
+                                    (1ULL << CAP_SYS_ADMIN) |
```

Comment:
mmh CAP_SYS_ADMIN is quite a big hammer, is it necessary? Isn't CAP_BPF enough?

Reply:
I get a denial with just `CAP_BPF`. To load a program I need `CAP_SYS_ADMIN` too.
I have this small PoC:
```
$ getcap sysctl_monitor
$ ./sysctl_monitor
libbpf: Failed to bump RLIMIT_MEMLOCK (err = -1), you might need to do it explicitly!
libbpf: Error in bpf_object__probe_loading():Operation not permitted(1). Couldn't load trivial BPF program. Make sure your kernel supports BPF (CONFIG_BPF_SYSCALL=y) and/or that RLIMIT_MEMLOCK is set to big enough value.
libbpf: failed to load object 'sysctl_monitor_bpf'
libbpf: failed to load BPF skeleton 'sysctl_monitor_bpf': -1
failed to open and load BPF object
$ sudo setcap cap_bpf=ep sysctl_monitor
$ ./sysctl_monitor
libbpf: prog 'sysctl_monitor': BPF program load failed: Operation not permitted
libbpf: prog 'sysctl_monitor': failed to load: -1
libbpf: failed to load object 'sysctl_monitor_bpf'
libbpf: failed to load BPF skeleton 'sysctl_monitor_bpf': -1
failed to open and load BPF object
$ sudo setcap cap_sys_admin=ep sysctl_monitor
$ ./sysctl_monitor
^C
$ echo $?
0
```
Maybe we can drop `CAP_SYS_ADMIN` just after BPF load (or failure or thereof)?

Reply:
yeah dropping it would be nice if it works

Reply:
I added a `drop_capability(CAP_SYS_ADMIN)` just after the program load, with a warning on failure as well